crates/libcontainer/src/tty.rs

```diff
@@ -89,36 +89,30 @@ pub fn setup_console_socket(
         linked: linked.to_path_buf().into(),
         console_socket_path: console_socket_path.to_path_buf().into(),
     })?;
-    // Using ManuallyDrop to keep the socket open.
-    let csocketfd = std::mem::ManuallyDrop::new(
-        socket::socket(
-            socket::AddressFamily::Unix,
-            socket::SockType::Stream,
-            socket::SockFlag::empty(),
-            None,
-        )
-        .map_err(|err| TTYError::CreateConsoleSocketFd { source: err })?,
-    );
-    let csocketfd = match socket::connect(
+    let csocketfd = socket::socket(
+        socket::AddressFamily::Unix,
+        socket::SockType::Stream,
+        socket::SockFlag::empty(),
+        None,
+    )
+    .map_err(|err| TTYError::CreateConsoleSocketFd { source: err })?;
+    socket::connect(
         csocketfd.as_raw_fd(),
         &socket::UnixAddr::new(linked.as_path()).map_err(|err| TTYError::InvalidSocketName {
             source: err,
             socket_name: socket_name.to_string(),
         })?,
-    ) {
-        Err(Errno::ENOENT) => -1,
```

Comment:
The logic of this branch (and checked by the `test_setup_console_socket_empty` test) is lost here.

Reply:
Yes, I don't quite get why is this error handled independently, Could you mind explain why do we have to return -1 when ENOENT?

Reply:
Well, How about returning error when the file is not present? I don't think it will succeed if the init process send the tty fd through a socket of fd -1.
It is difficult to modify it if this function return OwnedFd, I think we can't new an OwnedFd from a rawfd of -1, because it may panic if the `drop` of OwnedFd is called, and we will call `close` of an fd of -1.

Reply:
Can we get the background of why does it return fd of -1 here? It is a little wierd because I don't find any similar logic in runc.

Reply:
Hey so I went around in the git blame and tried to find out why this is specifically `-1` , but from what I see, the `-1` comes directly from the first commit (!) , and is kept as-is for no specific reason. Checking the code there, I think the intention was maybe to conform to the standard error policy (returning -1 on error) , but not sure. What I can see is that in that commit, this will just push error to a later time, when calling sendmsg with socket_fd of -1. Apart from that atleast I cannot find any reason in that or today's code why we specifically set it to -1 rather than return an error, because the code will error when it tries to use this raw fd for sendmsg. I think the correct way to proceed would be another analysis by someone else to check what I'm saying is right or not, and if it is, then replacing the -1 by the general error case.

@utam0k , correct me if I'm wrong here.

Reply:
I believe that it came from `railcar`. `-1` doesn't have a special reason. I'm afraid I've confused you... I'll note the person who first committed (that's me) 🫠
https://github.com/oracle/railcar/blob/ef5918e21e7ad9ffd25c1a507df96458ec4e0c24/src/main.rs#L539-L549

Reply:
So, Can we say that it is ok to return error rather than -1?

Reply:
@abel-von , yes I think we can just merge that into the error case.